### PR TITLE
Connectivity API: get information about links and transports through Session::info()

### PR DESCRIPTION
**Pull request by milyin**
implementation for https://github.com/eclipse-zenoh/zenoh/issues/692

- implemented transports() and links() methods
- implemented transport_events_listener() and link_events_listener() methods
- test added
- z_info example demonstrates new functionality
- adminspace uses the connectivity api, no separate implementation
- the adminspace keys `@/<local_zid>/session/transport/(multicast|unicast)/<remote_zid>` and `@/<local_zid>/session/transport/(multicast|unicast)/<remote_zid>/link/<link_id>` works as before, but uses the new API. The keeping the same adminspace behavior is ensured by test https://github.com/eclipse-zenoh/zenoh/pull/2334
- the priority and reliability for `Link` are none if transport doesn't support qos
- the reliability is derived from the link protocol if not configured explicitly

---
## 🏷️ Label-Based Checklist

Based on the labels applied to this PR, please complete these additional requirements:

**Labels:** `new feature`

## 🆕 New Feature Requirements

Since this PR adds a new feature:

- [x] **Feature scope documented** - Clear description of what the feature does and why it's needed
- [x] **Minimum necessary code** - Implementation is as simple as possible, doesn't overcomplicate the system
- [x] **New APIs well-designed** - Public APIs are intuitive, consistent with existing APIs
- [x] **Comprehensive tests** - All functionality is tested (happy path + edge cases + error cases)
- [x] **Examples provided** - Usage examples in code comments or separate example files
- [x] **Documentation added** - New docs explaining the feature, its use cases, and API
- [x] **Feature flag considered** - Can the feature be enabled/disabled for gradual rollout?
- [x] **Performance impact assessed** - Memory, CPU, storage implications measured
- [x] **Integration tested** - Feature works with existing features

**Consider:** Can this feature be split into smaller, incremental PRs?

**Instructions:**
1. Check off items as you complete them (change `- [ ]` to `- [x]`)
2. The PR checklist CI will verify these are completed

*This checklist updates automatically when labels change, but preserves your checked boxes.*

### CODE REVIEW

File: zenoh/src/api/session.rs [hunk 1]
```diff
@@ -2080,6 +2137,152 @@ impl SessionInner {
...
+        callback: Callback<TransportEvent>,
+        history: bool,
+    ) -> ZResult<Arc<TransportEventsListenerState>> {
+        let mut state = zwrite!(self.state);
+        let id = self.runtime.next_id();
+        trace!("declare_transport_events_listener_inner() => {id}");
+
+        let listener_state = Arc::new(TransportEventsListenerState { id, callback });
+
+        state
```
Comment (DenisBiryukov91) [inline zenoh/src/api/session.rs]: can we just use `zwrite!(self.state).transport_events_listeners.insert(id, listener_state.clone())` without the need to manually drop the lock ?

File: zenoh/src/api/session.rs [hunk 2]
```diff
@@ -2080,6 +2137,152 @@ impl SessionInner {
...
+        let id = self.runtime.next_id();
+        trace!("declare_transport_links_listener_inner() => {id}");
+
+        let listener_state = Arc::new(LinkEventsListenerState {
+            id,
+            callback,
+            transport_zid,
+        });
+
+        state
```
Comment (DenisBiryukov91) [inline zenoh/src/api/session.rs]: can we use `zwrite!(self.state).link_events_listeners.insert(id, listener_state.clone())` to avoid manually dropping the lock ?

File: zenoh/src/api/session.rs (hunk 1 above)
Comment (milyin) [reply zenoh/src/api/session.rs]: fixed

File: zenoh/src/api/session.rs (hunk 2 above)
Comment (milyin) [reply zenoh/src/api/session.rs]: fixed

File: zenoh/src/api/info.rs [hunk 3]
```diff
@@ -104,6 +120,347 @@ impl SessionInfo {
...
+    #[inline]
+    pub fn auth_identifier(&self) -> Option<&str> {
+        self.auth_identifier.as_deref()
+    }
+
+    /// Gets the optional priority range (min, max) associated with the link.
+    /// The numeric priority values corresponds [`Priority`](crate::qos::Priority)
+    /// but can also contain value 0 (Control) not exposed in the public enum.
+    #[inline]
+    pub fn priorities(&self) -> (u8, u8) {
```
Comment (OlivierHecart) [inline zenoh/src/api/info.rs]: Maybe `priorities()` and `reliability()` should return `None` if the associated transport has `is_qos() == false`.

File: zenoh/src/net/runtime/mod.rs [hunk 4]
```diff
@@ -145,6 +148,15 @@ pub trait IRuntime: Send + Sync {
     fn get_transports_unicast_peers(&self) -> Vec<TransportPeer>;
     fn get_transports_multicast_peers(&self) -> Vec<Vec<TransportPeer>>;
 
+    #[cfg(feature = "unstable")]
+    fn get_transports(&self) -> Box<dyn Iterator<Item = Transport> + Send + Sync>;
```
Comment (OlivierHecart) [inline zenoh/src/net/runtime/mod.rs:147]: While they coexist, new API and the old API should use the same internal calls to retrieve data. 
For example, the old API relies on `Runtime::get_transports_unicast_peers()` and `Runtime::get_transports_multicast_peers()` while the new API relies on `Runtime::get_transports()`. 
This is prone to incoherencies.

File: zenoh/src/net/runtime/mod.rs [hunk 5]
```diff
@@ -244,6 +256,49 @@ impl IRuntime for RuntimeState {
             .collect::<Vec<_>>()
     }
 
+    #[cfg(feature = "unstable")]
+    fn get_transports(&self) -> Box<dyn Iterator<Item = Transport> + Send + Sync> {
+        Box::new(
```
Comment (OlivierHecart) [inline zenoh/src/net/runtime/mod.rs]: Contrarily to the old API, it seems to me that the new API does not expose multicast transports.

**Comment by OlivierHecart** [comment]
Global comment:
It has been agreed to keep both old and new APIs for *Connectivity Status and Events* for now. This 2 APIs should return coherent infos as much as possible (it seems that some minor behaviors have been slightly changed in the new API). 
This should be achieved by making the 2 APIs use the same internal code as much as possible.

File: zenoh/src/api/info.rs (hunk 3 above)
Comment (milyin) [reply zenoh/src/api/info.rs]: Done

File: zenoh/src/net/runtime/mod.rs (hunk 4 above)
Comment (milyin) [reply zenoh/src/net/runtime/mod.rs:147]: Done, test verifying that old and new json matches added https://github.com/eclipse-zenoh/zenoh/pull/2334

File: zenoh/src/net/runtime/mod.rs (hunk 5 above)
Comment (milyin) [reply zenoh/src/net/runtime/mod.rs]: Fixed
